### PR TITLE
Unpickle Scala 2 types

### PR DESCRIPTION
**Pull request by sjrd**

### CODE REVIEW

File: shared/src/main/scala/tastyquery/reader/pickles/PickleReader.scala [hunk 1]
```diff
@@ -73,41 +80,80 @@ class PickleReader {
...
-      ???
+    def storeResultInEntries(result: MaybeExternalSymbol): Unit =
+      assert(entries(storeInEntriesAt) == null, entries(storeInEntriesAt))
+      entries(storeInEntriesAt) = result
+
+    def readExtSymbol(): MaybeExternalSymbol =
+      val name = readNameRef() // TODO .decode
+      val owner = if (atEnd) clsCtx.defn.RootPackage else readMaybeExternalSymbolRef()
+      name match
+        case nme.RootName        => clsCtx.moduleClassRoot
```
Comment (bishabosha) [inline shared/src/main/scala/tastyquery/reader/pickles/PickleReader.scala:127]:
```suggestion
        case nme.RootName        => defn.RootPackage
```

Comment (bishabosha) [reply shared/src/main/scala/tastyquery/reader/pickles/PickleReader.scala:127]: `moduleClassRoot` is the companion objects backing class

File: shared/src/main/scala/tastyquery/reader/pickles/PickleReader.scala [hunk 2]
```diff
@@ -73,41 +80,80 @@ class PickleReader {
...
+        case nme.RootPackageName => clsCtx.moduleRoot
```
[the remark this answers is not in the document]
Comment (bishabosha) [reply shared/src/main/scala/tastyquery/reader/pickles/PickleReader.scala:128]: `clsCtx.moduleRoot` is the companion object of the top level class

File: shared/src/main/scala/tastyquery/reader/pickles/PickleReader.scala [hunk 3]
```diff
@@ -132,27 +178,31 @@ class PickleReader {
...
         // newSymbol(owner, name1, flags1, localMemberUnpickler, privateWithin, coord = start)
-
-        // TODO: clsCtx at owner
-        clsCtx.createSymbol(name1, RegularSymbolFactory, addToDecls = false)
+        RegularSymbolFactory.createSymbol(name1, owner)
       case CLASSsym =>
         if isClassRoot then clsCtx.classRoot useWith { _.initialised = true }
         else if isModuleClassRoot then clsCtx.moduleClassRoot useWith { _.initialised = true }
-        else clsCtx.createSymbol(name.toTypeName, ClassSymbolFactory, addToDecls = false)
+        else ClassSymbolFactory.createSymbol(name.toTypeName, owner)
```
Comment (bishabosha) [inline shared/src/main/scala/tastyquery/reader/pickles/PickleReader.scala:197]: I guess when it is not a top level class we should still add to decls of owner?

**Review by bishabosha — CHANGES_REQUESTED** [review]
just the point about the root package mostly

File: shared/src/main/scala/tastyquery/reader/pickles/PickleReader.scala (hunk 3 above)
Comment (sjrd) [reply shared/src/main/scala/tastyquery/reader/pickles/PickleReader.scala]: Yes, indeed. Same as for Java classfiles inner classes. I added a comment.

File: shared/src/main/scala/tastyquery/reader/pickles/PickleReader.scala [hunk 4]
```diff
@@ -132,27 +178,31 @@ class PickleReader {
...
+        val tpe = readSymType()
+        sym.withDeclaredType(tpe)
       case MODULEsym =>
-        if isModuleRoot then {
-          clsCtx.moduleRoot
-        } else {
-          ???
-        }
+        if isModuleRoot then clsCtx.moduleRoot
+        else RegularSymbolFactory.createSymbol(name.toTermName, owner)
```
[the remark this answers is not in the document]
Comment (sjrd) [reply shared/src/main/scala/tastyquery/reader/pickles/PickleReader.scala]: Same as above: I added a comment.

**Review by bishabosha — APPROVED** [review]

**Pull request merged**